File: src/main/java/pt/joaomneto/titancompanion/adventurecreation/impl/fragments/sots/SOTSAdventureCreationSkillFragment.java
package pt.joaomneto.titancompanion.adventurecreation.impl.fragments.sots;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import pt.joaomneto.titancompanion.R;
import pt.joaomneto.titancompanion.adventurecreation.AdventureCreation;
import pt.joaomneto.titancompanion.adventurecreation.impl.SOTSAdventureCreation;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.AdapterView;
import android.widget.ListView;
import android.widget.SimpleAdapter;

public class SOTSAdventureCreationSkillFragment extends Fragment {

	public SOTSAdventureCreationSkillFragment() {

	}

	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		View rootView = inflater.inflate(R.layout.fragment_20sots_adventure_creation_skill, container, false);

		List<Map<String, String>> skillList = new ArrayList<Map<String,String>>();


		List<String> stringArray = new ArrayList<String>();
		for (SOTSMartialArt art : SOTSMartialArt.values()) {
			stringArray.add(getResources().getString(art.getNameId()));
		}

		for (String string : stringArray) {
			Map<String, String> skill = new HashMap<String, String>();
			skill.put("skill", string);
			skillList.add(skill);
		}

		SimpleAdapter mAdapter = new SimpleAdapter(getActivity(), skillList, R.layout.potions_item, new String[] {"skill"}, new int[] {R.id.potion_name} );

		ListView lView = (ListView) rootView.findViewById(R.id.skillList);
		lView.setAdapter(mAdapter);
		lView.setSelector(R.drawable.row_selector);

		lView.setOnItemClickListener(new AdapterView.OnItemClickListener() {

			@Override
			public void onItemClick(AdapterView<?> parent, final View view, int position, long id) {
				((SOTSAdventureCreation)getActivity()).setSkill(getArtFromString(stringArray.get(position)));
			}

		});

		return rootView;
	}

	@Override
	public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);

		view.findViewById(R.id.buttonSaveAdventure).setOnClickListener((View v) -> {
			((AdventureCreation) this.getActivity()).saveAdventure(v);
		});
	}

	SOTSMartialArt getArtFromString(String nameString) {
		for (SOTSMartialArt art : SOTSMartialArt.values()) {
			if (nameString.equals(getResources().getString(art.getNameId())))
				return art;
		}
		return null;
	}

}

File: src/main/java/pt/joaomneto/titancompanion/adventurecreation/impl/fragments/sots/SOTSAdventureCreationSkillFragment.kt
package pt.joaomneto.titancompanion.adventurecreation.impl.fragments.sots

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.ListView
import android.widget.SimpleAdapter
import pt.joaomneto.titancompanion.R
import pt.joaomneto.titancompanion.adventurecreation.AdventureCreation
import pt.joaomneto.titancompanion.adventurecreation.impl.SOTSAdventureCreation
import java.util.ArrayList
import java.util.HashMap

class SOTSAdventureCreationSkillFragment : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        super.onCreate(savedInstanceState)
        val rootView = inflater.inflate(R.layout.fragment_20sots_adventure_creation_skill, container, false)

        val skillList = ArrayList<Map<String, String>>()

        val stringArray = ArrayList<String>()
        for (art in SOTSMartialArt.values()) {
            stringArray.add(resources.getString(art.nameId))
        }

        for (string in stringArray) {
            val skill = HashMap<String, String>()
            skill["skill"] = string
            skillList.add(skill)
        }

        val mAdapter = SimpleAdapter(
            activity,
            skillList,
            R.layout.potions_item,
            arrayOf("skill"),
            intArrayOf(R.id.potion_name)
        )

        val lView = rootView.findViewById<ListView>(R.id.skillList)
        lView.adapter = mAdapter
        lView.setSelector(R.drawable.row_selector)

        lView.onItemClickListener = AdapterView.OnItemClickListener { parent, view, position, id ->
            (activity as SOTSAdventureCreation).setSkill(
                getArtFromString(stringArray[position])!!
            )
        }

        return rootView
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        view.findViewById<View>(R.id.buttonSaveAdventure).setOnClickListener { v: View -> (this.activity as AdventureCreation).saveAdventure() }
    }

    internal fun getArtFromString(nameString: String): SOTSMartialArt? {
        for (art in SOTSMartialArt.values()) {
            if (nameString == resources.getString(art.nameId))
                return art
        }
        return null
    }
}
